modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/metadata/DefaultComponentMetadataConfigurer.java

```diff
@@ -191,6 +209,10 @@ public <T extends ParameterizedDeclarer, D extends ParameterizedDeclaration> voi
 
   @Override
   public <T extends ComponentDeclaration> void configure(ParameterizedDeclaration<T> declaration) {
+    if (routesChainInputTypesResolvers.isEmpty() && chainInputTypeResolver != null
+        && declaration instanceof WithNestedComponentsDeclaration) {
+      applyChainInputTypeResolverToAllRoutes((WithNestedComponentsDeclaration<?>) declaration);
```

Comment:
what is this for? Let's avoid auto magical defaults. Also remember that the base line here is the SDK, not the core langauge

Reply:
SDK users won't use the configurer directly, but the SDK implementation should. Otherwise we would have repeated code on how to configure these things.

In either case, yes, this seems way too automagical to me

Reply:
Ok forget about the last paragraph, the annotation goes to the parameter, so they will need to add it to each route anyway.

Reply:
Resolved in https://github.com/mulesoft/mule/pull/13602/commits/43051b4b364026be6a1e9b5396f6648252260ec9